File: src/com/JavaSnakes/util/MenuCard.java
package com.JavaSnakes.util;

import java.awt.Component;
import java.awt.GridBagConstraints;
import java.awt.GridBagLayout;
import java.awt.Insets;
import javax.swing.JPanel;

public class MenuCard extends JPanel {

    public MenuCard() {
        setLayout(new GridBagLayout());
    }

    public void addInGrid(Component component, int y, int x) {
        addInGrid(component, y, x, 1);
    }

    public void addInGrid(Component component, int y, int x, int w) {
        addInGrid(component, y, x, w, null);
    }

    public void addInGrid(Component component, int y, int x, int w, Insets padding) {
        GridBagConstraints constraint = new GridBagConstraints();
        constraint.fill = GridBagConstraints.HORIZONTAL;
        constraint.gridy = y;
        constraint.gridx = x;
        constraint.gridwidth = w;
        if (padding != null) constraint.insets = padding;
        add(component, constraint);
    }
}

File: src/main/kotlin/com/JavaSnakes/util/MenuCard.kt
package com.javasnakes.util

import java.awt.Component
import java.awt.GridBagConstraints
import java.awt.GridBagLayout
import java.awt.Insets
import javax.swing.JPanel

class MenuCard : JPanel() {
    init {
        layout = GridBagLayout()
    }

    fun addInGrid(component: Component, y: Int, x: Int, w: Int = 1, padding: Insets? = null) {
        val constraint = GridBagConstraints()
        constraint.fill = GridBagConstraints.HORIZONTAL
        constraint.gridy = y
        constraint.gridx = x
        constraint.gridwidth = w
        if (padding != null) constraint.insets = padding
        add(component, constraint)
    }
}
